sqlglot/dialects/materialize.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+from sqlglot import exp
+from sqlglot.helper import seq_get
+from sqlglot.dialects.postgres import Postgres
+
+from sqlglot.tokens import TokenType
+from sqlglot.transforms import (
+    remove_unique_constraints,
+    ctas_with_tmp_tables_to_create_tmp_view,
+    preprocess,
+)
+import typing as t
+
+
+class Materialize(Postgres):
+    class Parser(Postgres.Parser):
+        NO_PAREN_FUNCTION_PARSERS = {
+            **Postgres.Parser.NO_PAREN_FUNCTION_PARSERS,
+            "MAP": lambda self: self._parse_map(),
+        }
+
+        def _parse_map(self) -> exp.ToMap:
+            if self._match(TokenType.L_PAREN):
+                to_map = self.expression(exp.ToMap, this=self._parse_select())
+                self._match_r_paren()
+                return to_map
+
+            if not self._match(TokenType.L_BRACKET):
+                self.raise_error("Expecting [")
+            entries = self._parse_csv(self._parse_map_entry)
+            if not self._match(TokenType.R_BRACKET):
+                self.raise_error("Expecting ]")
+            return self.expression(exp.ToMap, this=self.expression(exp.Struct, expressions=entries))
+
+        def _parse_map_entry(self) -> t.Optional[exp.PropertyEQ]:
+            key = self._parse_conjunction()
+            if not key:
+                return None
+            if not self._match(TokenType.FARROW):
+                self.raise_error("Expected =>")
+            value = self._parse_conjunction()
+            return self.expression(exp.PropertyEQ, this=key, expression=value)
+
+    class Generator(Postgres.Generator):
+        SUPPORTS_CREATE_TABLE_LIKE = False
+
+        TRANSFORMS = {
+            **Postgres.Generator.TRANSFORMS,
+            exp.AutoIncrementColumnConstraint: lambda self, e: "",
+            exp.Create: preprocess(
+                [
+                    remove_unique_constraints,
+                    ctas_with_tmp_tables_to_create_tmp_view,
+                ]
+            ),
+            exp.DataType: lambda self, e: self._datatype_sql(e),
+            exp.GeneratedAsIdentityColumnConstraint: lambda self, e: "",
+            exp.OnConflict: lambda self, e: "",
+            exp.PrimaryKeyColumnConstraint: lambda self, e: "",
+            exp.List: lambda self, e: self._list_sql(e),
```

Comment:
Since the parsing of LIST constructors is now handled in the base parser, do we want to move this to the base generator too? Perhaps we can even share code with array generation using the new `ARRAY_GENERATORS` constant?

Reply:
Possibly, yeah. Not sure how that'd look, but if we can adapt the existing logic for arrays and make minimal changes to support `LIST`, that sounds good to me.